engineapi/engineapi/actions.py

```diff
@@ -1491,6 +1505,130 @@ def list_leaderboards_resources(
     return query.all()
 
 
+def get_leaderboard_config_entry(
+    leaderboard_id: uuid.UUID,
+) -> BugoutSearchResult:
+    query = f"#leaderboard_id:{leaderboard_id}"
+    configs = bc.search(
+        token=MOONSTREAM_ADMIN_ACCESS_TOKEN,
+        journal_id=MOONSTREAM_LEADERBOARD_GENERATOR_JOURNAL_ID,
+        query=query,
+        limit=1,
+    )
+
+    results = cast(List[BugoutSearchResult], configs.results)
+
+    if len(configs.results) == 0 or results[0].content is None:
+        raise LeaderboardConfigNotFound(
+            f"Leaderboard config not found for {leaderboard_id}"
+        )
+
+    return results[0]
+
+
+def get_leaderboard_config(
+    leaderboard_id: uuid.UUID,
+) -> Dict[str, Any]:
+    """
+    Return leaderboard config from leaderboard generator journal
+    """
+
+    entry = get_leaderboard_config_entry(leaderboard_id)
+
+    content = json.loads(entry.content)  # type: ignore
+
+    if "status:active" not in entry.tags:
+        content["leaderboard_auto_update_active"] = False
+    else:
+        content["leaderboard_auto_update_active"] = True
+
+    return content
+
+
+def update_leaderboard_config(
+    leaderboard_id: uuid.UUID, config: LeaderboardConfigUpdate
+) -> Dict[str, Any]:
+    """
+    Update leaderboard config in leaderboard generator journal
+
+    """
+
+    entry_config = get_leaderboard_config_entry(leaderboard_id)
+
+    current_config = LeaderboardConfig(**json.loads(entry_config.content))  # type: ignore
+
+    new_params = config.params
+
+    for key, value in new_params.items():
+        if key not in current_config.params:
+            continue
+
+        current_config.params[key] = value
+
+    # we replace values of parameters that are not None
+
+    entry = bc.update_entry_content(
+        token=MOONSTREAM_ADMIN_ACCESS_TOKEN,
+        journal_id=MOONSTREAM_LEADERBOARD_GENERATOR_JOURNAL_ID,
+        title=entry_config.title,
+        entry_id=entry_config.entry_url.split("/")[-1],
+        content=json.dumps(current_config.dict()),
+    )
+
+    new_config = json.loads(entry.content)
+
+    if "status:active" not in entry.tags:
+        new_config["leaderboard_auto_update_active"] = False
+    else:
+        new_config["leaderboard_auto_update_active"] = True
+
+    return new_config
+
+
+def activate_leaderboard_config(
```

Comment:
It seems a bit odd to check for only "status:active". Presumably someone could think it's correct to set "status:inactive". I suppose if we just tell everyone to use the API to update the config it will be fine.

Reply:
Only the Moonstream API and our team has access to that journal. Moonstream users do not have access to the journal interface. Their only access is through this API.